test/metafields_test.go

```diff
@@ -0,0 +1,196 @@
+	Id        *int                   `json:"id,omitempty"`
+	Name      *string                `json:"name,omitempty"`
+	Scope     *models.MetafieldScope `json:"scope,omitempty"`
+	DataCount *int                   `json:"data_count,omitempty"`
+	InputType *string                `json:"input_type,omitempty"`
+	Enum      []string               `json:"enum"`
+}
+
+func (s *APISuite) TestMetafields() {
+	ctx := context.Background()
+
+	customer := s.createCustomer(ctx)
+	productFamilyID := s.createProductFamily(ctx)
+	product := s.createProduct(ctx, *productFamilyID.Id)
+	subscription := s.newSubscription(customer, product, "", []models.CreateSubscriptionComponent{})
+
+	radioFieldName := s.fkr.RandomStringWithLength(20)
+	dropdownFieldName := s.fkr.RandomStringWithLength(20)
+	textFieldName := s.fkr.RandomStringWithLength(20)
+
+	resp, err := s.client.SubscriptionsController().CreateSubscription(ctx, &models.CreateSubscriptionRequest{
+		Subscription: subscription,
+	})
+	s.NoError(err)
+	s.Equal(http.StatusCreated, resp.Response.StatusCode)
+
+	subs := resp.Data.Subscription
+
+	cases := []struct {
+		name         string
+		resourceType models.ResourceType
+		metafields   interface{}
+		assert       func(t *testing.T, resp models.ApiResponse[[]models.Metafield], err error)
+	}{
+		{
+			name:         "subscriptions",
+			resourceType: models.ResourceType_SUBSCRIPTIONS,
+			metafields: []models.Metafield{
+				{
+					Name:      &dropdownFieldName,
+					InputType: strPtr("dropdown"),
+					Scope: &models.MetafieldScope{
+						PublicShow: toPtr[models.IncludeOption](models.IncludeOption_INCLUDE),
+						PublicEdit: toPtr[models.IncludeOption](models.IncludeOption_INCLUDE),
+					},
+					Enum: models.NewOptional[interface{}](interfacePtr([]string{
+						"option 1",
+						"option 2",
+					})),
+				},
+				{
+					Name: &textFieldName,
+				},
+			},
+			assert: func(t *testing.T, resp models.ApiResponse[[]models.Metafield], err error) {
+				s.NoError(err)
+				s.Contains([]int{http.StatusOK, http.StatusCreated}, resp.Response.StatusCode)
+
+				dropdownField := resp.Data[0]
+				s.Equal(dropdownFieldName, *dropdownField.Name)
+				s.Equal("dropdown", *dropdownField.InputType)
+				// s.Len(dropdownField.Enum.Value(), 2) unusable at this stage
+				s.Equal(models.IncludeOption_INCLUDE, *dropdownField.Scope.PublicShow)
+				s.Equal(models.IncludeOption_INCLUDE, *dropdownField.Scope.PublicEdit)
+
+				textField := resp.Data[1]
+				s.Equal(textFieldName, *textField.Name)
+				s.Equal("text", *textField.InputType)
+
+				r, err := s.client.CustomFieldsController().CreateMetadata(
+					ctx,
+					models.ResourceType_SUBSCRIPTIONS,
+					fmt.Sprintf("%d", *subs.Id),
+					&models.CreateMetadataRequest{
+						Metadata: []models.CreateMetadata{
+							{
+								Name:  dropdownField.Name,
+								Value: strPtr("option 1"),
+							},
+							{
+								Name:  textField.Name,
+								Value: strPtr("something"),
+							},
+						},
+					})
```

Comment:
Could we check any response properties to test the deserialization? Length of the response array (2) and some properties like `value` and `resource_id` may be enough

Reply:
done